epidose/back_end/deploy.yml

```diff
@@ -44,44 +47,41 @@
     - debug:
          msg: "{{ gunicorn_debug.stdout_lines|list }}"
 
-    - name: remove default nginx site
+    - name: create nginx config file for ha-server
       become: yes
       become_user: root
       file:
-        path: /etc/nginx/sites-available/ha-server.conf
+        path: /etc/nginx/conf.d/{{ server_name }}.conf
         state: touch
         owner: root
         group: root
         mode: 0644
 
-    - name: add ha-server configurations
+    - name: add nginx configurations
+      tags: nginx
       become: yes
       become_user: root
       blockinfile: |
-        dest=/etc/nginx/sites-available/ha-server.conf
+        dest=/etc/nginx/conf.d/{{ server_name }}.conf
         content='server {
                     listen 80;
```

Comment:
Shouldn't this be the HTTPS port? Also, if you don't listen on 80, certificate renewal becomes much easier.

Reply:
The HTTPS port is added automatically by the Certbox.
Please see the corresponding configuration file after the execution of the Ansible script:

```
# BEGIN ANSIBLE MANAGED BLOCK
server {
listen 80;
server_name epidose.ellak.gr;
root /home/epidose/epidose/epidose/back_end;
access_log /var/log/nginx_access.log;
error_log /var/log/nginx_error.log;

if ($scheme != "https") {
return 301 https://epidose.ellak.gr$request_uri;
}

location / {
proxy_set_header X-Forward-For $proxy_add_x_forwarded_for;
proxy_set_header Host $http_host;
proxy_redirect off;
proxy_pass http://185.70.186.134:5010;
}

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/epidose.ellak.gr/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/epidose.ellak.gr/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
# END ANSIBLE MANAGED BLOCK
```

For port 80, if invoked, a user will get a 301 Moved permanently message.
I am not sure if the renewal becomes easy or not.
But, I will try to remove `listen  80;`, run the Ansible script again, and see what happens.

Reply:
I have tested the deploy script with the requested changes and even if I do not add the `listen 80` port, the certbot adds it:

```
    listen 80; # managed by Certbot

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/epidose.ellak.gr/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/epidose.ellak.gr/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
```

Also, this [article](https://letsencrypt.org/docs/allow-port-80/) from let's encrypt states that it is not a problem.
But I can definitely remove the line after the configurations are set.
What do you think?